File: practice-contest/ABC086C-Traveling.py
#!/usr/bin/env python3


def f(txy):
    pt, px, py = txy[0]
    
    if pt < (px + py) or sum([pt % 2, px % 2, py % 2]) % 2:
        print("No")
        return
    
    else:
        for t, x, y in txy[1:]:
            mvt, mvx, mvy = t - pt, x - px, y - py
            if mvt < (mvx + mvy) or sum([mvt % 2, mvx % 2, mvy % 2]) % 2:
                print("No")
                return
            pt, px, py = t, x, y
        
        print("Yes")


N = int(input())
txy = [tuple(int(i) for i in input().split(" ")) for _ in range(N)]
f(txy)

File: practice-contest/ABC086C-Traveling.c
#include <stdio.h>

int main(void)
{
	int n;
	scanf("%d", &n);

	int t[n], x[n], y[n];
	for (int i = 0; i < n; ++i) scanf("%d %d %d", &t[i], &x[i], &y[i]);
	
	int pt, px, py;
	pt = t[0];
	px = x[0];
	py = y[0];

	if (pt < (px + py) | ((pt % 2) + (px % 2) + (py % 2)) % 2)
	{
		printf("No");
		return 0;
	}

	int it, ix, iy, mt, mx, my;
	for (int j = 1; j < n; ++j)
	{
		it = t[j];
		ix = x[j];
		iy = y[j];

		mt = it - pt;
		mx = ix - px;
		my = iy - py;

		
		if (mt < (mx + my) | ((mt % 2) + (mx % 2) + (my % 2)) % 2)
		{
			printf("No");
			return 0;
		}

		pt = mt;
		px = mx;
		py = my;
	}

	printf("Yes");
	return 0;
}

